### PR TITLE
fix: fix warning on mixed (non)pointer receivers

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

The template mixed pointer and non-pointer receivers and that causes some warnings

### CODE REVIEW

File: tools/config/go-templates/model_simple.mustache [hunk 1]
```diff
@@ -172,14 +172,15 @@ func (o *{{classname}}) Set{{name}}(v {{vendorExtensions.x-go-base-type}}) {
 
 {{/required}}
 {{/vars}}
-func (o {{classname}}) MarshalJSONWithoutReadOnly() ([]byte, error) {
+func (o *{{classname}}) MarshalJSONWithoutReadOnly() ([]byte, error) {
```
Comment (gssbzn) [inline tools/config/go-templates/model_simple.mustache:175]: this is the change, the rest is regenerating things

**Review by lantoli — APPROVED** [review]

File: tools/config/go-templates/model_simple.mustache (hunk 1 above)
Comment (lantoli) [reply tools/config/go-templates/model_simple.mustache:175]: i suppose we prefer this because performance?

Comment (gssbzn) [reply tools/config/go-templates/model_simple.mustache:175]: you need pointers to modify the struct, which we do for setters, and you should not mix pointers and non pointers so stick to the one you need

Screenshot 2024-10-01 at 11 53 18

Comment (lantoli) [reply tools/config/go-templates/model_simple.mustache:175]: thanks, i didn't think of the setters

**Pull request merged**